src/nrnpython/nrnpy_nrn.cpp

```diff
@@ -1499,17 +1499,17 @@ static PyObject* NPySecObj_insert(NPySecObj* self, PyObject* args) {
         PyObject* tpyobj;
         if (PyArg_ParseTuple(args, "O", &tpyobj)) {
             auto _tpyobj_tracker = nb::borrow(tpyobj);
+            auto self = nb::borrow((PyObject*) raw_self);
             // Returned object to be discarded
-            auto out_o = nb::steal(PyObject_CallMethod(tpyobj, "insert", "O", (PyObject*) self));
+            auto out_o = nb::steal(PyObject_CallMethod(tpyobj, "insert", "O", self.ptr()));
             if (!out_o.is_valid()) {
                 PyErr_Clear();
                 PyErr_SetString(
                     PyExc_TypeError,
                     "insert argument must be either a string or an object with an insert method");
                 return nullptr;
             }
-            Py_INCREF(self);
-            return (PyObject*) self;
+            return self.release().ptr();
```

Comment:
Since we're holding the reference count at `+1`, we can `.release()` the `self` to prevent the `DECREF` in the dtor of `self`.

The effect is that we return a "new reference" for `raw_self` as prescribed by the API documentation for `PyCFunction`.